Public/Src/Cache/ContentStore/Distributed/NuCache/TransitioningContentLocationStore.cs

```diff
@@ -271,21 +277,34 @@ public Task<BoolResult> UnregisterAsync(Context context, IReadOnlyList<ContentHa
 
             var operationContext = new OperationContext(context);
 
+            // Ideally, we want to remove content we know won't be used again for quite a while. We don't have that
+            // information, so we instead have an evictability metric. Here we obtain and sort by that evictability
+            // metric.
             var pageSize = _configuration.EvictionWindowSize;
+            var poolMultiplier = _configuration.MaximumEvictionPoolMultiplier;
+
+            float? takeoutFraction = null;
+            if (contentHashesWithInfo.Count > 0)
+            {
+                // We want to remove roughly the size of the pool divided by the amount of content at each iteration
+                takeoutFraction = 1 / contentHashesWithInfo.Count;

```

Comment:
Also, I don't think this is what we want takeout fraction to be, even without the integer division issue. Every time we pull in a page, we want to return a fraction (pool count) / (cache count) of the pool, i.e. (pool count)^2 / (cache count) items. First off, this fraction is not fixed for the call duration of the call into QueryAndOrderInPages, but rather increases as the pool count increases from (page size) to (pool maximum). [This may disagreement may be related to the current code that fills the pool before returning anything.] Second, even if the fraction were fixed, it would be (pool size) / (cache count), not 1.0 / (cache count).